src/ChangeStream.php

```diff
@@ -94,10 +98,15 @@ public function next()
     {
         $resumable = false;
         try {
+            if ($this->valid()) {
+                $this->previousValid = true;
```

Comment:
I'd propose `$hasAdvanced` as a property name. Like "is", the prefix implies that it's a boolean type, and I think it better matches the purpose of the variable as a flag that is initially `false` and is switched once to `true` (where it remains thereafter).

Also, I think we can move this to the existing `if ($this->valid())` branch after `$this->csIt->next();`.

Assuming you implement [PHPLIB-335](https://jira.mongodb.org/browse/PHPLIB-335) in the same PR, I think it'd make sense for `$this->hasAdvanced = true` to occur between the conditional key increment and resume token extraction. For example:

```php
$this->csIt->next();
if ($this->valid()) {
    if ($this->hasAdvanced) {
        $this->key++;
    }
    $this->hasAdvanced = true;
    $this->resumeToken = $this->extractResumeToken($this->csIt->current());
}
```

In the spirit of PHPLIB-335, this will ensure that `$hasAdvanced` flag is toggled even if resume token extraction fails.